src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -188,5 +255,155 @@ private bool AssemblyContainsTagHelpers(string assemblyName, IReadOnlyList<TagHe
+
+            public override void Visit(RazorSyntaxTree tree)
+            {
+                _source = tree.Source;
+                Visit(tree.Root);
+            }
+
+            public override void VisitRazorDirective(RazorDirectiveSyntax node)
+            {
+                var descendantLiterals = node.DescendantNodes();
+                foreach (var child in descendantLiterals)
+                {
+                    if (!(child is CSharpStatementLiteralSyntax literal))
+                    {
+                        continue;
+                    }
+
+                    var context = literal.GetSpanContext();
+                    if (context == null)
+                    {
+                        // We can't find a chunk generator.
+                        continue;
+                    }
+                    else if (context.ChunkGenerator is AddTagHelperChunkGenerator addTagHelper)
+                    {
+                        // Make sure this node exists in the file we're parsing and not in its imports.
+                        if (_filePath.Equals(_source.FilePath, StringComparison.Ordinal))
+                        {
+                            addTagHelper.Diagnostics.Add(
+                                ComponentDiagnosticFactory.Create_UnsupportedTagHelperDirective(node.GetSourceSpan(_source)));
+                        }
+                    }
+                    else if (context.ChunkGenerator is RemoveTagHelperChunkGenerator removeTagHelper)
+                    {
+                        // Make sure this node exists in the file we're parsing and not in its imports.
+                        if (_filePath.Equals(_source.FilePath, StringComparison.Ordinal))
+                        {
+                            removeTagHelper.Diagnostics.Add(
+                                ComponentDiagnosticFactory.Create_UnsupportedTagHelperDirective(node.GetSourceSpan(_source)));
+                        }
+                    }
+                    else if (context.ChunkGenerator is TagHelperPrefixDirectiveChunkGenerator tagHelperPrefix)
+                    {
+                        // Make sure this node exists in the file we're parsing and not in its imports.
+                        if (_filePath.Equals(_source.FilePath, StringComparison.Ordinal))
+                        {
+                            tagHelperPrefix.Diagnostics.Add(
+                                ComponentDiagnosticFactory.Create_UnsupportedTagHelperDirective(node.GetSourceSpan(_source)));
+                        }
+                    }
+                    else if (context.ChunkGenerator is AddImportChunkGenerator usingStatement && !usingStatement.IsStatic)
+                    {
+                        // Get the namespace from the using statement. Split it at '=' in case there is an alias.
+                        var @namespace = usingStatement.ParsedNamespace.Split('=').LastOrDefault();
```

Comment:
Doing a split at `=` here so we don't barf on seeing a using with an alias. This isn't enough to actually support aliases.

Reply:
So if we don't support aliases we should skip it right?

`using Foo = MyApp.Bar` doesn't bring types in `MyApp` or `MyApp.Bar` into scope.

Reply:
Good catch. Yeah, the right thing to do would be to completely ignore any usings with alias.